### PR TITLE
Applyied patch, fixed compilation errors and formatted the code

### PR DESCRIPTION
**Pull request by REASY**
I've left things which are unclear for me with `// FIXME ?!`

This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3084)

### CODE REVIEW

**Comment by REASY** [comment]
test now!

File: src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala [hunk 1]
```diff
@@ -1277,6 +1278,7 @@ object ChoosesMode {
         RoutingResponse.dummyRoutingResponse
       },
       parkingResponse = if (withParking) {
+//        Some(ParkingInquiryResponse(ParkingStall.lastResortStall(boundingBox,costInDollars = 0), 0))
```
Comment (REASY) [inline src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala:1281]: Remove commented code

File: src/main/scala/beam/agentsim/agents/ridehail/DefaultRideHailDepotParkingManager.scala [hunk 2]
```diff
@@ -0,0 +1,729 @@
...
+  * @param parkingFilePath
+  * @param valueOfTime
+  * @param tazTreeMap
+  * @param random
+  * @param boundingBox
+  * @param distFunction
+  * @param parkingStallCountScalingFactor
+  * @param beamServices
+  */
+class DefaultRideHailDepotParkingManager[GEO: GeoLevel](
```
Comment (REASY) [inline src/main/scala/beam/agentsim/agents/ridehail/DefaultRideHailDepotParkingManager.scala:67]: @dimaopen please, take a look to this class. It is a replacement of `RideHailDepotParkingManager`

File: src/main/scala/beam/agentsim/agents/ridehail/DefaultRideHailDepotParkingManager.scala [hunk 3]
```diff
@@ -0,0 +1,729 @@
...
+      boundingBox,
+      distFunction
+    )
+
+  ParkingZoneFileUtils.toCsv(
+    rideHailParkingZones,
+    outputDirectory.getOutputFilename(DefaultRideHailDepotParkingManager.outputRidehailParkingFileName)
+  )
+
+  val mnlMultiplierParameters: Map[ParkingMNL.Parameters, UtilityFunctionOperation] = Map(
```
Comment (REASY) [inline src/main/scala/beam/agentsim/agents/ridehail/DefaultRideHailDepotParkingManager.scala:136]: This manager is used for CAVs  (_robots_), so we don't need to pass `mnlMultiplierParameters`, it is created here

**Comment by haitamlaarabi** [comment]
test now!

**Comment by haitamlaarabi** [comment]
test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
test!

**Review by JustinPihony — COMMENTED** [review]
Partial review I started - so wanted to at least get this out there.

File: src/main/scala/beam/agentsim/agents/ridehail/RideHailAgent.scala [hunk 4]
```diff
@@ -265,8 +343,10 @@ class RideHailAgent(
       )
       eventsManager.processEvent(new PersonEntersVehicleEvent(tick, Id.createPersonId(id), vehicle.id))
       val isTimeForShift = shifts.isEmpty || shifts.get
-        .exists(shift => shift.lowerBound <= tick && shift.upperBound >= tick)
+        .find(shift => shift.range.lowerBound <= tick && shift.range.upperBound >= tick)
```
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/ridehail/RideHailAgent.scala:346]: I don't see why this was changed to use the less efficient version - it should remain an exists.

**Review by JustinPihony — COMMENTED** [review]
Nothing egregious - most comments are helpful items and not showstoppers. However there are a number of functional changes - this is not just refactorings. Many seem as though they will heavily conflict with https://github.com/LBNL-UCB-STI/beam/pull/2895. If you want an analysis of functional changes then that will take much more time - so it will rely on what level of trust you have with the supplier of the code.

**Comment by JustinPihony** [comment]
Test!

**Comment by JustinPihony** [comment]
@haitamlaarabi  ( or @wrashid ?) Beyond the open questions, my mention of clashes with the charging PR, and a number of other functional changes - this is at least passing tests. What are the thoughts on where this is headed (merging soon, need further review, etc)?

Or @wrashid

**Comment by REASY** [comment]
Commit e36cc30345f11f5a7e0d58a71272aa6cf3ab53a2 is needed to improve performance a bit
image

**Comment by REASY** [comment]
Commit 1134ca7229c5305b1006eea2a390dd865184e1b0 is needed to improve performance due to resize
image

**Comment by REASY** [comment]
Commit 9963987502f18accdb5c8c069f79e3109f6f8d42  is needed to improve performance of `getRepositioningVehicles`
image

**Comment by REASY** [comment]
Commit f6a85da891b904e28b62a2e964489074f7ed52a1 is needed to improve performance of  `computeTravelTimeAndDistanceAndCost` which is called very often (> 100kk times for production smart-baseline)
image

**Comment by REASY** [comment]
test!

**Comment by REASY** [comment]
test!

**Comment by REASY** [comment]
test!

**Comment by REASY** [comment]
test now!

**Comment by REASY** [comment]
Summary of optimizations:
image

**Review by haitamlaarabi — APPROVED** [review]

**Pull request merged**